src/util/ogr_source_util.cpp

```diff
@@ -447,41 +449,125 @@ bool OGRSourceUtil::readAttributesIntoCollection(AttributeArrays &attributeArray
 }
 
 // initializes the column index for the time columns
-void OGRSourceUtil::initTimeReading(OGRFeatureDefn *attributeDefn)
+void OGRSourceUtil::initTimeReading(OGRFeatureDefn *attributeDefn, OGRLayer *layer, const QueryRectangle &rect)
 {
 	if(timeSpecification == TimeSpecification::NONE)
 		return;
 
+	time1Type = OGRFieldType::OFTInteger;
+    time2Type = OGRFieldType::OFTInteger;
 	time1Index = -1;
 	time2Index = -1;
 
 	for(int i = 0; i < attributeDefn->GetFieldCount(); i++)
 	{
 		OGRFieldDefn *fieldDefn = attributeDefn->GetFieldDefn(i);
 		std::string name(fieldDefn->GetNameRef());
-		if(name == time1Name)
+		if(name == time1Name){
+			time1Type = fieldDefn->GetType();
 			time1Index = i;
-		else if(name == time2Name)
+		}
+		else if(timeSpecification != TimeSpecification::START && name == time2Name){
+			time2Type = fieldDefn->GetType();
 			time2Index = i;
+		}
 	}
 
 	if(time1Index == -1)
 		throw OperatorException("OGR Source: time1 attribute not found.");
-	
+
 	if(timeSpecification == TimeSpecification::START_DURATION || timeSpecification == TimeSpecification::START_END)
 	{
 		if(time2Index == -1)
-			throw OperatorException("OGR Source: time1 attribute not found.");
+			throw OperatorException("OGR Source: time2 attribute not found.");
 	}
+
+    //try filtering the attributes via OGRLayer, so we don't have to do it manually in readTimeIntoCollection
+	timeAlreadyFiltered = trySettingTemporalFilter(layer, rect);
+}
+
+bool OGRSourceUtil::trySettingTemporalFilter(OGRLayer *layer, const QueryRectangle &rect){
+	// DateTime and Date are the supported field data type!
+	// Both fields have to available and of type DateTime/Date, because only START_END as TimeSpecification is supported.
+	// That's because I couldn't find a way to express in the sql like query (time_attribute + time_duration).
+	if(timeSpecification != TimeSpecification::START_END)
+        return false;
+
+    // can try to set the filter without the fields being of type Date/DateTime. But that is unsafe and
+    // can result in empty collections. So the filter could be set on string data fields and still be a good query.
+    // but both attributes still have to be present and be of a Date/DateTime structure (as a string).
+    bool hasSaveTypes = (time1Type == OGRFieldType::OFTDateTime || time1Type == OGRFieldType::OFTDate) &&
+                           (time2Type == OGRFieldType::OFTDateTime || time2Type == OGRFieldType::OFTDate);
```

Comment:
is it safe to have mixed date and datetime for start and end?

Reply:
Yes, it is tested. Both types are always compared with DateTimeStrings.